kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigConfigMap.java

```diff
@@ -306,19 +326,99 @@ private static GenericKubernetesResource filterResourceFromRef(VirtualKafkaClust
                 .orElseThrow(() -> filterResourceNotFound(cluster, filterRef));
     }
 
-    private static VirtualCluster getVirtualCluster(VirtualKafkaCluster cluster,
-                                                    KafkaClusterRef kafkaClusterRef,
-                                                    ProxyIngressModel ingressModel) {
-
-        ProxyIngressModel.VirtualClusterIngressModel virtualClusterIngressModel = ingressModel.clusterIngressModel(cluster).orElseThrow();
-        String bootstrap = kafkaClusterRef.getSpec().getBootstrapServers();
-        return new VirtualCluster(
-                ResourcesUtil.name(cluster), new TargetCluster(bootstrap, Optional.empty()),
-                null,
-                Optional.empty(),
-                virtualClusterIngressModel.gateways(),
-                false, false,
-                filterNamesForCluster(cluster));
+    private static Optional<VirtualCluster> getVirtualCluster(Context<KafkaProxy> context,
+                                                              VirtualKafkaCluster cluster,
+                                                              KafkaClusterRef kafkaClusterRef,
+                                                              ProxyIngressModel ingressModel) {
+        try {
+            ProxyIngressModel.VirtualClusterIngressModel virtualClusterIngressModel = ingressModel.clusterIngressModel(cluster).orElseThrow();
+            String bootstrap = kafkaClusterRef.getSpec().getBootstrapServers();
+            return Optional.of(new VirtualCluster(
+                    ResourcesUtil.name(cluster),
+                    new TargetCluster(bootstrap, buildTargetClusterTls(context, cluster)),
+                    null,
+                    Optional.empty(),
+                    virtualClusterIngressModel.gateways(),
+                    false, false,
+                    filterNamesForCluster(cluster)));
+        }
+        catch (InvalidClusterException e) {
+            SharedKafkaProxyContext.addClusterCondition(context, cluster, e.accepted());
+        }
+        return Optional.empty();
+    }
+
+    @NonNull
+    private static Optional<Tls> buildTargetClusterTls(Context<KafkaProxy> context,
+                                                       VirtualKafkaCluster cluster) {
+        return Optional.ofNullable(cluster.getSpec())
+                .map(VirtualKafkaClusterSpec::getTargetCluster)
+                .map(io.kroxylicious.kubernetes.api.v1alpha1.virtualkafkaclusterspec.TargetCluster::getTls)
+                .map(io.kroxylicious.kubernetes.api.v1alpha1.virtualkafkaclusterspec.targetcluster.Tls::getCertificateRef)
+                .map(certRef -> {
+                    return new Tls(new KeyPair(
```

Comment:
Are we happy to only support PEM certificates initally? They're what cert-manager will generate by default, though you can coax it into populating its output `Secrets` with jks and p12 files. But I'd rather not expend effort into trying to support all the formats, at least until we're able to support TLS on all the connections.